### PR TITLE
Add latitude and longitude to mf2->AS object conversion

### PR DESCRIPTION
**Pull request by kylewm**
Converts .h-entry > .p-location.h-geo to a proper ActivityStreams object with a location
that has latitude and longitude.

In the AS1 spec, location doesn't really seem to be an "object" per se, but it seems
like a convenient simplification to treat it as one

### CODE REVIEW

**Comment by snarfed** [comment]
first step toward snarfed/bridgy#317? cool!

> In the AS1 spec, location doesn't really seem to be an "object" per se, but it seems

really? http://activitystrea.ms/specs/json/schema/activity-schema.html#location seems to say it is. AS2 too: http://www.w3.org/TR/activitystreams-vocabulary/#dfn-location . maybe some of the confusion is that AS has multiple specs, the "platonic ideal" schema/vocab and the various serializations (json, atom, etc). whee!

**Comment by kylewm** [comment]
oh man, thank you for pointing that out. I guess I saw that "objectType" was missing and jumped to conclusions. Need to actually read the spec, not just look at examples :P

File: granary/microformats2.py [hunk 1]
```diff
@@ -293,6 +295,17 @@ def json_to_object(mf2):
     'tags': [json_to_object(cat) for cat in props.get('category', [])],
   }
 
+  lat, lng = prop.get('latitude'), prop.get('longitude')
+  if lat and lng:
+    try:
+      lat, lng = float(lat), float(lng)
+      # 2 leading places for latitude, 3 for longitude
+      obj['position'] = '%0+10.6f%0+11.6f/' % (lat, lng)
```
Comment (snarfed) [inline granary/microformats2.py]: you can actually leave this blank; it already gets populated in `Source.postprocess_object()`. your formatting is great though, feel free to add it there if you want!

**Comment by snarfed** [comment]
lgtm otherwise!

File: granary/microformats2.py (hunk 1 above)
Comment (kylewm) [inline granary/microformats2.py]: nice! sorry I jumped the gun on merging

Comment (snarfed) [inline granary/microformats2.py]: np!

Comment (kylewm) [inline granary/microformats2.py]: oh actually, I think this is only called when converting silos to AS. Should I add a call to `postprocess_activity` in `microformats2.html_to_activities`?

Comment (snarfed) [inline granary/microformats2.py]: definitely, please do!

Comment (kylewm) [inline granary/microformats2.py]: blerg sorry I hadn't looked closely enough. microformats2 is not a class, so it's not obvious how to call postprocess_activity without making the method static which I don't think is a good idea.

I propose just leaving out "position" and adding a TODO for now... :shrug:

Comment (snarfed) [inline granary/microformats2.py]: make `postprocess_object` (and/or `_activity`) a class method and call them like that? looks like we currently override them in `googleplus`, but those could be class methods too. (the `self.actor_name()` call in `postprocess_activity` is misleading, it's actually a static method.)

alternatively you're welcome to just add the TODO and keep the `position` code here. up to you!

**Pull request merged**